src/Core/Settings.h

```diff
@@ -972,7 +972,6 @@ class IColumn;
     M(Bool, output_format_bson_string_as_string, false, "Use BSON String type instead of Binary for String columns.", 0) \
     M(Bool, input_format_bson_skip_fields_with_unsupported_types_in_schema_inference, false, "Skip fields with unsupported types while schema inference for format BSON.", 0) \
     \
-    M(Bool, regexp_dict_allow_other_sources, false, "Allow regexp_tree dictionary to use sources other than yaml source.", 0) \
     M(Bool, format_display_secrets_in_show_and_select, false, "Do not hide secrets in SHOW and SELECT queries.", IMPORTANT) \
```

Comment:
Please also adjust the corresponding comment about "regexp_dict_allow_other_sources" in the [docs.](https://clickhouse.com/docs/en/sql-reference/dictionaries)

Reply:
Kind of related: Speaking of documentation: I just put myself into the shoes of someone who doesn't know regexp tree dictionaries already and read the [docs](https://clickhouse.com/docs/en/sql-reference/dictionaries) again ... they are unfortunately in no good shape.

First, the regexp dict docs really belong below https://clickhouse.com/docs/en/sql-reference/dictionaries#ways-to-store-dictionaries-in-memory.

> Regexp Tree dictionary stores multiple trees of regular expressions with attributions. Users can retrieve strings in the dictionary. If a string matches the root of the regexp tree, we will collect the corresponding attributes of the matched root and continue to walk the children. If any of the children matches the string, we will collect attributes and rewrite the old ones if conflicts occur, then continue the traverse until we reach leaf nodes.

I'd really start with a small motivating example (e.g. URL matching) and then explain how regexp tree dictionaries solve it + why this approach is better than other dictionary types. Ideally, tree traversal is explained using the example, otherwise sentences like "rewrite the old ones if conflicts occur" remain too abstract and confusing at this point.

> We only allow YAMLRegExpTree to work with regexp_tree dicitionary layout.

Typo: `dicitionary`. More importantly, this should rather sound sth like that: `RegExp Tree dictionaries are defined using a yaml file containing the regular expression tree, called "YAMLRegExpTree".`

In the example of the yaml config: Please use yaml comments to paraphrase the regexps as plain English. Readers don't like to switch to [here](https://github.com/google/re2/wiki/Syntax) just to understand the example. Please also add a sentence what this example is about. At this point the use case is not clear.

> The name and version is user-defined attributions in the dicitionary. 

`is` --> `are`, Typo: `dicitionary`. Also, it is still not clear here what term `attribution` means. Maybe `attribute` is a better term.

> The versions (which can be any name that not appear in attributions or the key) indicates the children nodes of this tree.

Very confusing. Why not `children`?

In the example:

>       version: '10'

Needs explanation what `10` means in the example.

> Back Reference
> The value of an attribution could contain a back reference which refers to a capture group of the matched regular expression. Reference number ranges from 1 to 9 and writes as $1 or \1.

Really needs examples, unclear what is meant otherwise.

Reply:
Good suggestion, I will think about how to write a good doc. We can solve the doc in another pr with `how to use it on the cloud`